### PR TITLE
Rl modularize

### PR DESCRIPTION
**Pull request by RLuckom**
First refactor into profiles.

### CODE REVIEW

File: broad-wb-malkov.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  "dns": {
+    "default": {
+      "name": "dsp-techops",
+      "zone": "dsp-techops.broadinstitute.org"
+    }
+  },
+  "terraform_state": {
+    "default": {
+      "state_path_prefix":"dsp-techops",
```
Comment (hjfbynara) [inline broad-wb-malkov.json]: while this won't work - I think it is blurring things.  I think we should put all individual state in object path that has nothing to do with dsp-techops.  When I did my first env the path prefix was "workbench"  and then I added a terra env identifier.  I think we should go ahead and use the "workbench" object path and decide what the next level should be.

File: broad-wb-perf2.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  "dns": {
+    "default": {
+      "name": "dsp-techops",
+      "zone": "dsp-techops.broadinstitute.org"
+    }
+  },
+  "terraform_state": {
+    "default": {
+      "state_path_prefix":"dsp-techops",
```
Comment (hjfbynara) [inline broad-wb-perf2.json]: same comment as above.

File: broad-wb-rluckom.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  },
+  "dns": {
+    "default": {
+      "name": "dsp-techops",
+      "zone": "dsp-techops.broadinstitute.org"
+    }
+  },
+  "terraform_state": {
+    "default": {
+      "state_path_prefix":"dsp-techops",
```
Comment (hjfbynara) [inline broad-wb-rluckom.json]: ditto

File: dev.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  },
+  "regions": {
+    "default": {
+      "name": "us-central1",
+      "zone_name": "us-central1-a"
+    }
+  },
+  "dns": {
+    "default": {
+      "name": "dsp-techops",
```
Comment (hjfbynara) [inline dev.json]: ditto

**Comment by hjfbynara** [comment]
Since you got rid of the larger terra-env-api-services.tf that enable all the service apis for the entire infrastructure - in lieu of what I see as individual profile enabling just the services that profile needs - I think we are now missing some apis that are kind of important.   Some are specific to a terra service (ie thurloe needs pubsub, sqladmin, sql-component), but some like logging, monitoring iamcredentials,... are global to the entire terra env.

**Comment by hjfbynara** [comment]
Related to comment above - if you are going to delegate api-service enabling to the terra service profile - thurloe will need somethat that enables the cloudsql and pubsub apis.

**Comment by RLuckom** [comment]
If there are APIs that we need but no profile needs them, then there's stuff we're not putting in terraform that we should move into terraform at some point. We should use the API as the way to document and start that work. For instance, if we need the iamcredentials API to run terra but nothing we're deploying here needs it, then there's _something_ outside terraform (and outside the deployment) that requires it. We should address that by making a profile for whatever that thing is, even if right now the only thing it does is enable the services it requires.

**Comment by gmalkov** [comment]
I guess I agree in general that the logic to grant API access and set up whatever uses that access should be in the same place, but I don't know if that's a reasonable restriction to put on ourselves for the perf environment MVP. Making lightweight profiles that just grant API access as an outline for future work seems fairly reasonable though.

**Pull request merged**